backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -112,8 +112,14 @@ CREATE TABLE blocks(
 
 -- Every transaction on chain.
 CREATE TABLE transactions(
-    -- Index of the transaction within the block.
+    -- Global index of the transaction, i.e. across blocks.
+    -- This is used to quickly determine counts of transactions, so
+    -- this _must_ be strictly increasing without skipping any values.
     index
```

Comment:
Hmm how would I define the next `total_transaction_count` value? Wouldn't I need to do a `COUNT(*)` over the `transactions` table with a `WHERE` filter for the block's index? Not that it would be a big or expensive count, but a +1 for every transaction row is simpler in that sense.

Reply:
We have all the transactions part of the block in memory when we prepare the block, so there it could just count and store the transaction count for each block in the prepared block.
When 'saving' we already have the `BlockProcessingContext` which you could extend to track the total transaction count of the previously inserted block and use that to compute the new total when inserting.